pkg/util/cluster/cluster.go

```diff
@@ -394,22 +405,39 @@ func (c *Cluster) SetupWorkloadIdentity(ctx context.Context, vnetResourceGroup s
 		},
 	)
 
-	for _, wi := range platformWorkloadIdentityRoles {
-		c.log.Infof("creating WI: %s", wi.OperatorName)
-		resp, err := c.msiClient.CreateOrUpdate(ctx, vnetResourceGroup, wi.OperatorName, armmsi.Identity{
-			Location: to.StringPtr(c.Config.Location),
-		}, nil)
+	newIdentities := []*armmsi.Identity{}
+	if c.Config.IdentityPoolResourceGroup != "" {
+		c.log.Info("Claiming workload identities")
+		pool := NewManagedIdentityPool(c.msiClient, c.Config.IdentityPoolResourceGroup)
+		timeout := time.Duration(c.Config.IdentityPoolClaimDurationHours) * time.Hour
+		identities, err := pool.ClaimIdentities(ctx, len(platformWorkloadIdentityRoles), c.Config.VnetResourceGroup, c.Config.ClusterName, timeout)
```

Comment:
Do we need to do `len(platformWorkloadIdentityRoles) + 1` here, to account for the cluster MSI?